File: backend/app/schemas/crop.py
from pydantic import BaseModel
from typing import Optional, List

class CropBase(BaseModel):
    name: str
    type: str
    season: str
    duration: str
    water_needs: str
    expected_yield: str
    status: str = "actief"
    icon: str = "🌱"
    description: Optional[str] = None
    soil_temp: Optional[str] = None
    soil_type: Optional[str] = None
    sunlight: Optional[str] = None
    tips: Optional[str] = None

class CropCreate(CropBase):
    pass

class CropUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[str] = None
    season: Optional[str] = None
    duration: Optional[str] = None
    water_needs: Optional[str] = None
    expected_yield: Optional[str] = None
    status: Optional[str] = None
    icon: Optional[str] = None
    description: Optional[str] = None
    soil_temp: Optional[str] = None
    soil_type: Optional[str] = None
    sunlight: Optional[str] = None
    tips: Optional[str] = None

class Crop(CropBase):
    id: int

    class Config:
        from_attributes = True

File: frontend/src/types/crop.ts
export interface Crop {
  id: number;
  name: string;
  type: string;
  season: string;
  duration: string;
  water_needs: string;
  expected_yield: string;
  status: 'actief' | 'inactief';
  icon: string;
  description?: string;
  soil_temp?: string;
  soil_type?: string;
  sunlight?: string;
  tips?: string;
}
